lib/SIL/SILValue.cpp

```diff
@@ -58,56 +58,67 @@ void ValueBase::replaceAllUsesWithUndef() {
     return inst;
-  // TODO: MultiValueInstruction
+  if (auto *result = dyn_cast<MultipleValueInstructionResult>(this))
+    return result->getParent();
   return nullptr;
 }
 
 Optional<ValueBase::DefiningInstructionResult>
 ValueBase::getDefiningInstructionResult() {
-  if (auto inst = dyn_cast<SingleValueInstruction>(this))
-    return DefiningInstructionResult{ inst, 0 };
-  // TODO: MultiValueInstruction
+  if (auto *inst = dyn_cast<SingleValueInstruction>(this))
+    return DefiningInstructionResult{inst, 0};
+  if (auto *result = dyn_cast<MultipleValueInstructionResult>(this))
+    return DefiningInstructionResult{result->getParent(), result->getIndex()};
   return None;
 }
 
 SILBasicBlock *SILNode::getParentBlock() const {
-  auto *NonConstThis = const_cast<SILNode *>(this);
-  if (auto *Inst = dyn_cast<SILInstruction>(NonConstThis))
+  auto *CanonicalNode =
+      const_cast<SILNode *>(this)->getCanonicalSILNodeInObject();
+  if (auto *Inst = dyn_cast<SILInstruction>(CanonicalNode))
     return Inst->getParent();
-  // TODO: MultiValueInstruction
-  if (auto *Arg = dyn_cast<SILArgument>(NonConstThis))
+  if (auto *Arg = dyn_cast<SILArgument>(CanonicalNode))
     return Arg->getParent();
   return nullptr;
 }
 
 SILFunction *SILNode::getFunction() const {
-  auto *NonConstThis = const_cast<SILNode *>(this);
-  if (auto *Inst = dyn_cast<SILInstruction>(NonConstThis))
+  auto *CanonicalNode =
+      const_cast<SILNode *>(this)->getCanonicalSILNodeInObject();
+  if (auto *Inst = dyn_cast<SILInstruction>(CanonicalNode))
     return Inst->getFunction();
-  // TODO: MultiValueInstruction
-  if (auto *Arg = dyn_cast<SILArgument>(NonConstThis))
+  if (auto *Arg = dyn_cast<SILArgument>(CanonicalNode))
     return Arg->getFunction();
   return nullptr;
 }
 
 SILModule *SILNode::getModule() const {
-  auto *NonConstThis = const_cast<SILNode *>(this);
-  if (auto *Inst = dyn_cast<SILInstruction>(NonConstThis))
+  auto *CanonicalNode =
+      const_cast<SILNode *>(this)->getCanonicalSILNodeInObject();
+  if (auto *Inst = dyn_cast<SILInstruction>(CanonicalNode))
     return &Inst->getModule();
-  // TODO: MultiValueInstruction
-  if (auto *Arg = dyn_cast<SILArgument>(NonConstThis))
+  if (auto *Arg = dyn_cast<SILArgument>(CanonicalNode))
     return &Arg->getModule();
   return nullptr;
 }
 
 const SILNode *SILNode::getCanonicalSILNodeSlowPath() const {
-  assert(getStorageLoc() != SILNodeStorageLocation::Instruction &&
-         hasMultipleSILNodes(getKind()));
-  return &static_cast<const SILInstruction &>(
-            static_cast<const SingleValueInstruction &>(
-              static_cast<const ValueBase &>(*this)));
+  assert(getStorageLoc() != SILNodeStorageLocation::Instruction);
+
+  if (isa<SingleValueInstruction>(this)) {
+    assert(hasMultipleSILNodes(getKind()));
+    return &static_cast<const SILInstruction &>(
+        static_cast<const SingleValueInstruction &>(
+            static_cast<const ValueBase &>(*this)));
```

Comment:
I think the old indentation pattern looked nicer.

Reply:
I just did git-clang-format.